dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+    };
+
+    match args.command {
+        DeployCommand::Stop => teardown_a4x2(&sh, &env)?,
+        DeployCommand::Start(_) | DeployCommand::RunTests(_) => {
+            if !env.in_ci {
+                // Teardown previous deploy if it exists, before wiping the data
+                // for it. If this errors, we assume the deploy doesn't exist
+                // and carry on.
+                let result = teardown_a4x2(&sh, &env);
+                eprintln!("teardown result: {:?}", result);
+                eprintln!("continuing regardless of whether there were errors");
+
+                // Delete any results from previous runs. We don't mind if
+                // there's errors. This needs to run as root in case there are
+                // artifacts owned by root left around from the deploy.
+                cmd!(sh, "pfexec rm -rf").arg(&env.work_dir).run()?;
+            }
+
+            // Create work dir
+            fs::create_dir_all(&env.out_dir)?;
+            fs::create_dir_all(&env.work_dir)?;
+            sh.change_dir(&env.work_dir);
+
+            unpack_a4x2(&sh, &env)?;
+            prepare_to_launch_a4x2(&sh, &env)?;
+
+            // Launch a4x2. This can fail. In fact it fails quite a bit...
+            // enough that perhaps it's owed at least one retry? But we do not
+            // retry right now.
+            //
+            // We capture the error so we can tear down a4x2 gracefully. Note
+            // that in the case of running tests, we will produce an error if
+            // launching a4x2 succeeds but the tests themselves fail.
+            let result =
+                try_launch_a4x2(&sh, &env).and_then(|_| match &args.command {
+                    DeployCommand::RunTests(test_args) => {
+                        run_tests(&sh, &env, &test_args)
+                    }
+                    _ => Ok(()),
+                });
+
+            // TODO: Collect evidence!!! Try to retrieve the sled-agent logs
+            // (and perhaps others?), to help diagnose a4x2 launch failures, and
+            // provide context alongside live tests.
+
+            // If we are in CI, we do not need to tear down, because the system
+            // state is about to be wiped from existence anyway.
+            if !env.in_ci {
+                // Outside of CI, we do not tear down if we are just running the
+                // start command, or if the user requested to leave a4x2 up
+                // after running the tests.
+                match args.command {
+                    DeployCommand::RunTests(RunTestsArgs {
+                        leave_running: false,
+                        ..
+                    }) => {
+                        // We ignore any error here, because the result we
+                        // actually want to produce is whether tests passed
+                        let _ = teardown_a4x2(&sh, &env);
+                    }
```

Comment:
I don't think we should ignore this error.  Either **teardown_a4x2()** should work, or we need to stop and figure out why it did not.